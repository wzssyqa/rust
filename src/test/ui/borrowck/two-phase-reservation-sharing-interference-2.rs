// compile-flags: -Z borrowck=mir -Z two-phase-borrows

// This is similar to two-phase-reservation-sharing-interference.rs
// in that it shows a reservation that overlaps with a shared borrow.

fn main() {
    let mut v = vec![0, 1, 2];
    let shared = &v;

    v.push(shared.len()); //~ ERROR cannot borrow `v` as mutable

    assert_eq!(v, [0, 1, 2, 3]);
}
