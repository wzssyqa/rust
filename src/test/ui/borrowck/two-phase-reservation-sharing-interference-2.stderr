error[E0502]: cannot borrow `v` as mutable because it is also borrowed as immutable
  --> $DIR/two-phase-reservation-sharing-interference-2.rs:10:5
   |
LL |     let shared = &v;
   |                  -- immutable borrow occurs here
LL | 
LL |     v.push(shared.len()); //~ ERROR cannot borrow `v` as mutable
   |     ^      ------ immutable borrow later used here
   |     |
   |     mutable borrow occurs here

error: aborting due to previous error

For more information about this error, try `rustc --explain E0502`.
